generate/generate_nircam.py

```diff
@@ -441,3 +405,94 @@
     ignore_attributes = 'XIdlVert1 XIdlVert2 XIdlVert3 XIdlVert4 YIdlVert1 YIdlVert2 YIdlVert3 YIdlVert4'.split()
     ignore_attributes += ([s for s in DISTORTION_ATTRIBUTES if 'Idl2Sci' in s])
     compare.compare_siaf(new_siaf, reference_siaf_input=ref_siaf, fractional_tolerance=1e-6, ignore_attributes=ignore_attributes)
+
+
+######################################
+# Print corners
+######################################
+
+goodlist  = ['NRCA2_MASK210R','NRCA5_MASK335R','NRCA5_MASK430R','NRCA4_MASKSWB','NRCA5_MASKLWB',
+             'NRCA2_TAMASK210R','NRCA2_FSTAMASK210R','NRCA5_TAMASK335R','NRCA2_FSTAMASK210R','NRCA5_TAMASK430R',
+             'NRCA5_FSTAMASK430R','NRCA4_TAMASKSWB','NRCA4_TAMASKSWBS','NRCA4_FSTAMASKSWB','NRCA5_TAMASKLWB','NRCA5_TAMASKLWBL','NRCA5_FSTAMASKLWB']
+
+in_file = os.path.join(JWST_SOURCE_DATA_ROOT,instrument,'nircam_siaf_aperture_definition.txt')
+out_file = os.path.join(JWST_DELIVERY_DATA_ROOT, instrument,'nircam_enhanced_siaf_aperture_definition.txt')
+
+print('Writing SIAF aperture definition file including subarrays and V2/V3 of reference point')
+with open(in_file) as fp:
+    lines = fp.readlines()
+
+
+with open(out_file, 'w') as the_file:
```

Comment:
If what I mention in my main comment is true, this code will only be executed once and therefore does need to be part of the repo.

Reply:
We decided not to alter the current format of the nircam_siaf_aperture_definition.txt file, to avoid compatibility issues (also with other subsystems).
We would like the "enhanced" file to be produced every time a change needs to be introduced to nircam apertures. In this respect, the enhanced file has the same function as the "JIRA" plots you are already producing in generate_nircam. Please let me know if you'd like this piece of code to be moved to a different location within generate_nircam

Reply:
As for the ._* files...sorry will try not to '''git add *''' next time

Reply:
The only customer of nircam_siaf_aperture_definition.txt in pysiaf, we we can add columns there without any problems.

Reply:
Ok...let me rephrase that. The subarray corners, the way I defined this enhanced file, are output products, computed self-consistently in pysiaf. we want to use them to double check entries in the PRD subarray file, that's why we don't want to have them in input in the regular siaf aperture deifnition file....at least for now (we may want to have all info in a single file in the future, or at least we are discussing that). Moreover, although pysiaf might be the only customer, the nircam team has already shared the enhanced file with OSS for them to verify, e.g. that their v2/v3 ref positions in the TA scripts are consistent with what pysiaf computes. I think we really prefer having this enhnaced file be separated from the source file and avoid confusion.

Reply:
Our current workflow implies that we purge the pre_delivery_data directory before we make a pysiaf release. Is this compatible with your use cases?

Reply:
I believe so. Basically I'd like that file to be treated like the "JIRA plots": 

1. we can produce that file ourselves by running generate_nircam and looking at the pre_delivery products, to make sure that whatever we put in a pysiaf PR looks consistent to us
2. if that's ok with you, that file could be part of the package you guys ask us to verify before merging a PR (i.e. the one you attach to the JSOCINT issue)
3.  just like the aperture plots in that package, the file itself does not need to be included in a release

For these reasons I believe it may make more sense to move the piece of code that creates the enhanced file inside the ```if emulate_delivery``` statement.  My new proposal is:

- I could package those ~60 lines of code into a separate function within the pysiaf.utils module and call it ```create_enhanced_aperture_file```

- I could place an ```if enhanced_aperture_file:  create_enhanced_aperture_file```
 statement after the ```for compare_to in [pysiaf.JWST_PRD_VERSION]:``` loop and before the ```    # run some tests on the new SIAF``` one 

Let me know what you think

Reply:
Thank you for clearing up our questions. 
I like your suggested implementation. 

@shanosborne , do you agree, too?

Reply:
My new commit should address this last point

Reply:
I like the change of putting the new code under the the `if emulate_delivery` statement, it clearly groups this new files with all the other analysis files. Looks good to me!